user-operator/src/main/java/io/strimzi/operator/user/model/KafkaUserModel.java

```diff
@@ -178,15 +181,24 @@ public void maybeGenerateCertificates(CertManager certManager, PasswordGenerator
                 String caCrt = userSecret.getData().get("ca.crt");
                 String userCrt = userSecret.getData().get("user.crt");
                 String userKey = userSecret.getData().get("user.key");
+                String userKeyStore = userSecret.getData().get("user.p12");
+                String userKeyStorePassword = userSecret.getData().get("user.password");
                 if (originalCaCrt != null
                         && originalCaCrt.equals(caCrt)
                         && userCrt != null
                         && !userCrt.isEmpty()
                         && userKey != null
-                        && !userKey.isEmpty()) {
+                        && !userKey.isEmpty()
+                        && userKeyStore != null
```

Comment:
How does this work after upgrade? When the old secrets will not have the P12 and password files?

Reply:
good point, just to be clear I guess you are talking about upgrading from Strimzi 0.14.0 to 0.15.0 (for example) where a `KafkaUser` secret already exists but it doesn't have P12 and password, right? In that case, we should generate P12 and password from the existing key and cert. It means that the check on null and empty for P12 and password doesn't make sense here. At this point, I have a question why did you check null and empty for key, crt, ... ? Is it possible that a `KafkaUser' secret for TLS-related user exist but it doesn't contain what it should contain?

Reply:
Yeah, I meant the secrets created before this PR which which will not have the p12 and password files. Actually, the same applies to the server secrets from the previous PR as well and the broker secrets etc. We cannot regenerate the keys just to add the p12.

Reply:
Good point. I will fix here generating just P12 and password in that case. I will work on a different PR for fixing the case of CA certs and broker certs Secrets.
Just about my question above, what was the scenario of testing not null and not empty cert, key etc etc when a KafkaUser secret was already in place?

Reply:
I'm not sure what do you mean.

Reply:
I meant ... we are in the `if` when the Secret already exists. After retrieving content from the Secret, there is the `if` checking that `userCrt != null` and `!userCrt.isEmpty()` etc etc.
Why it was checked? What's the scenario when you retrieve a user secret for a TLS user and those are null or empty?

Reply:
Always can someone come and change it.

Reply:
Ah, so it was more a "defensive" programming thing :-)